tests/api_test.py

```diff
@@ -1847,6 +1851,7 @@ def f(tup):
     jtu._check_dtypes_match(actual, desired)
     jtu.check_eq(actual, desired)
 
+  @jax.numpy_dtype_promotion('standard')  # Test explicitly exercises implicit dtype promotion.
```

Comment:
Same here: test explicitly is testing heterogeneous inputs, so I don't think we should change that.